cls/SourceControl/Git/Production.cls

```diff
@@ -0,0 +1,373 @@
+        } else {
+            // Special case for Config Item Settings PTD, requires checking PTD CDATA for Item and Class name
+            set settingsPTDText = $zconvert(deployDoc.GetValueAt("/Export/Document[2]/1"),"I","XML")
+            set settingsPTD = ##class(EnsLib.EDI.XML.Document).%New(settingsPTDText)
+            set itemClass = settingsPTD.GetValueAt("/Item/@ClassName")
+            set itemName = settingsPTD.GetValueAt("/Item/@Name")
+            set internalName = ..CreateInternalName(productionName, itemName, itemClass, 0)
+        }
+    }
+}
+
+/// Given an internal name for a PTD item, returns a suggested filename for export, as well as:
+/// - itemName: name of the configuration item
+/// - productionName: name of the associated production
+/// - isProdSettings: if true, this item is a production settings; if false, this item is a configuration item settings
+ClassMethod ParseInternalName(internalName, noFolders As %Boolean = 0, Output fileName, Output itemName, Output productionName, Output isProdSettings As %Boolean)
+{
+    set name = $piece(internalName,".",1,*-1)
+    if 'noFolders {
+        set name = $replace(name,"||","/")
+        set $ListBuild(productionName, name) = $ListFromString(name, "/")
+    }
+    // Abbreviate "ProductionSettings" to "ProdStgs", "Settings" to "Stgs".
+    Set prefix = $Case($Extract(name), "P":"ProdStgs-", "S":"Stgs-", :"")
+    set isProdSettings = ($Extract(name) = "P")
+    Set name = $Piece(name,"-",2,*)
+    set $ListBuild(itemName, itemClassName) = $ListFromString(name, "|")
+    set name = prefix_$select(
+        $get(itemClassName) '= "": itemName_$zhex($zcrc(itemClassName,6)),
+        1: name
+    )
+    if 'noFolders {
+        set name = productionName _ "/" _ name
+    }
+    set fileName = $translate($replace(name, ".", "_") _ ".xml", "\", "/")
+}
+
+/// Calculates the internal name for a decomposed production item
+ClassMethod CreateInternalName(productionName = "", itemName = "", itemClassName = "", isProductionSettings As %Boolean = 0)
+{
+    return $select(
+        isProductionSettings: productionName_"||ProductionSettings-"_productionName_".PTD",
+        1: productionName _ "||Settings-" _ itemName _ "|" _ itemClassName _ ".PTD"
+    )
+}
+
+/// Given an external name for a PTD item, removes that item from the production.
+ClassMethod RemoveItemByExternalName(externalName, nameMethod) As %Status
+{
+    set sc = $$$OK
+    set productionName = $replace($piece($replace(externalName,"\","/"),"/",*-1),"_",".")
+    set production = ##class(Ens.Config.Production).%OpenId(productionName,,.sc)
+    $$$QuitOnError(sc)
+    set itemToRemove = $$$NULLOREF
+    for i=1:1:production.Items.Count() {
+        set configItem = production.Items.GetAt(i)
+        set itemInternalName = ..CreateInternalName(productionName, configItem.Name, configItem.ClassName)
+        set itemExternalName = $classmethod($$SrcCtrlCls^%buildccr, nameMethod, itemInternalName)
```

Comment:
Better to use ##class(%Studio.SourceControl.Interface).SourceControlClassGet() - maybe wrap in a macro